src/components/SignInForm/index.jsx

```diff
@@ -2,31 +2,60 @@ import React from "react";
 import { useForm } from "react-hook-form";
 import Joi from "joi";
 import { joiResolver } from "@hookform/resolvers/joi";
+import { useTranslation } from "react-i18next";
 import InputErrorMessage from "../InputErrorMessage";
 import { getErrorClass } from "../../utils/formErrorHelpers";
-import "./index.scss";
 
-const schema = Joi.object({
-  email: Joi.string()
-    .required()
-    .email({ tlds: { allow: ["com", "net", "edu"] } })
-    .messages({
-      "string.empty": `Please provide your email`,
-      "string.email": `Please provide a valid email`,
-    }),
-  password: Joi.string().required().min(8).max(64).strict().messages({
-    "string.empty": `Please provide a password`,
-  }),
-});
+import "./index.scss";
 
 export default function SignInForm({ submit }) {
-  const { register, handleSubmit, errors, reset } = useForm({
+  const { t } = useTranslation();
+
+  const schema = Joi.object({
+    email: Joi.string()
+      .required()
+      .email({ tlds: { allow: ["com", "net", "edu"] } })
+      .messages({
+        "string.empty": `${t("signIn.validationMessage.emptyEmail")}`,
+        "string.email": `${t("signIn.validationMessage.emailNotValid")}`,
+      }),
+    password: Joi.string()
+      .required()
+      .min(8)
+      .max(64)
+      .strict()
+      .messages({
+        "string.empty": `${t("signIn.validationMessage.emptyPassword")}`,
+        "string.min": `${t("signIn.validationMessage.passwordMin")}`,
+        "string.max": `${t("signIn.validationMessage.passwordMax")}`,
+      }),
+  });
+
+  const { register, handleSubmit, errors, reset, setError } = useForm({
     resolver: joiResolver(schema),
   });
 
-  const onSubmit = (data) => {
-    submit(data);
-    reset();
+  const onSubmit = async (data) => {
+    const status = await submit(data);
+    switch (status) {
+      case "auth/wrong-password":
+        setError("password", {
+          type: "Incorrect password",
+          message: `${t("signIn.validationMessage.incorrectPassword")}`,
+        });
+        break;
+      case "auth/user-not-found":
+        setError("email", {
+          type: "Email not found",
```

Comment:
translations

Reply:
I should not translate the type, as this will be inside the code and other part of code will check it in order to display the related message (which will be displayed).